test/shopify-cli/theme/theme_admin_api_throttler/bulk_job_test.rb

```diff
@@ -0,0 +1,197 @@
+              method: "PUT",
+              api_version: "unstable",
+              body: JSON.generate({
+                assets: [
+                  { key: "file1.txt", value: "file1" },
+                ]
+              })
+            ).returns(
+              [
+                207,
+                {
+                  "results" => [
+                    {
+                      "code" => 200,
+                      "body" => {
+                        "asset" => {
+                          "key" => "file1.txt",
+                        }
+                      }
+                    },
+                  ],
+                },
+                {}
+              ]
+            )
+
+          @ctx.expects(:puts).with("12:30:59 {{green:Synced }} {{>}} {{blue:update file1.txt}}").once
+
+          time_freeze do
+            @job.perform!
+          end
+        end
+
+        def test_perform_when_bulk_request_error
+          @job = BulkJob.new(bulk)
+          ShopifyCLI::AdminAPI
+            .expects(:rest_request)
+            .with(
+              @ctx,
+              shop: @theme.shop,
+              path: "themes/#{@theme.id}/assets/bulk.json",
+              method: "PUT",
+              api_version: "unstable",
+              body: JSON.generate({
+                assets: [
+                  { key: "file1.txt", value: "file1" },
+                ]
+              })
+            ).returns(
+              [
+                400,
+                {},
+                {}
+              ]
+            )
+          @job.expects(:handle_requeue).once
+          @job.perform!
+        end
+
+        def test_perform_when_asset_update_error
+          @job = BulkJob.new(bulk)
+          ShopifyCLI::AdminAPI
+            .expects(:rest_request)
+            .with(
+              @ctx,
+              shop: @theme.shop,
+              path: "themes/#{@theme.id}/assets/bulk.json",
+              method: "PUT",
+              api_version: "unstable",
+              body: JSON.generate({
+                assets: [
+                  { key: "file1.txt", value: "file1" },
+                ]
+              })
+            ).returns(
+              [
+                207,
+                {
+                  "results" => [
+                    {
+                      "code" => 422,
+                      "body" => {
+                        "errors" => {
+                          "asset" => "Something is wrong with this file!!!",
+                        }
+                      }
+                    },
+                  ],
+                },
+                {}
+              ]
+            )
+          # TODO: determine how to handle
+        end
+
+        private
+
+        def bulk
+          @bulk ||= stub(
+            "Bulk",
+            ready?: true,
+            consume_put_requests: [generate_put_request("file1", 34_021)],
+            admin_api: @admin_api,
+          )
+        end
+
+        def simulate_operations
```

Comment:
Once this short block is written, should just be able to pass block to stubbed requests and check correct I/O